File: Plugins/Aspose_Barcode_Java_for_PHP/src/aspose/barcode/WorkingWithBarcodeRecognition/AdvancedBarcodeRecognitionFeatures/SwitchBarcodeRecognitionModes.php
<?php
namespace Aspose\Barcode\WorkingWithBarcodeRecognition\AdvancedBarcodeRecognitionFeatures;

use com\aspose\barcoderecognition\BarCodeReadType as BarCodeReadType;
use com\aspose\barcoderecognition\BarCodeReader as BarCodeReader;
use com\aspose\barcoderecognition\RecognitionMode as RecognitionMode;
use com\aspose\barcoderecognition\ManualHint as ManualHint;

class SwitchBarcodeRecognitionModes{

    public static function run($dataDir=null){

        $img = $dataDir . "barcode.jpg";

        # initialize barcode reader
        $barcode_reader_type = new BarCodeReadType();
        $reader = new BarCodeReader($img, $barcode_reader_type->Code39Standard);

        # Set recognition mode
        $recognitionMode=new RecognitionMode();
        $reader->setRecognitionMode($recognitionMode->ManualHints);

        # Set manual hints
        $manualHint=new ManualHint();
        $reader->setManualHints($manualHint->InvertImage);
        $reader->setManualHints($manualHint->IncorrectBarcodes);

        # Call read method
        while (java_values($reader->read())) {
            print "Barcode CodeText: " . (string)$reader->getCodeText().PHP_EOL;
        }

        # Close reader
        $reader->close();
    }
}

File: Plugins/Aspose-BarCode-Java-for-Jython/asposebarcode/WorkingWithBarcodeRecognition/AdvancedBarcodeRecognitionFeatures/SwitchBarcodeRecognitionModes.py
from asposebarcode import Settings
from com.aspose.barcoderecognition import BarCodeReadType
from com.aspose.barcoderecognition import BarCodeReader
from com.aspose.barcoderecognition import RecognitionMode
from com.aspose.barcoderecognition import ManualHint

class SwitchBarcodeRecognitionModes:

    def __init__(self):
        dataDir = Settings.dataDir + 'WorkingWithBarcodeRecognition/AdvancedBarcodeRecognitionFeatures/SwitchBarcodeRecognitionModes/'
        
        img = dataDir + "barcode.jpg"

        # initialize barcode reader
        barcode_reader_type = BarCodeReadType
        reader = BarCodeReader(img, barcode_reader_type.Code39Standard)

        # Set recognition mode
        recognitionMode = RecognitionMode
        reader.setRecognitionMode(recognitionMode.ManualHints)

        # Set manual hints
        manualHint = ManualHint
        reader.setManualHints(manualHint.InvertImage)
        reader.setManualHints(manualHint.IncorrectBarcodes)

        # Call read method
        while (reader.read()):
            print "Barcode CodeText: " + reader.getCodeText()
       

        # Close reader
        reader.close()

if __name__ == '__main__':        
    SwitchBarcodeRecognitionModes()